CakePHP 4.x compatibility

swiffer opened the pull request:
- code style
- type hints

swiffer committed:
bump

ADmad commented on src/TestSuite/Fixture/ChecksumTestFixture.php:
```diff
@@ -15,43 +17,43 @@
+     *
+     * @var array
+     */
+    public static $tableHashes = [];
```
Does this property need to be public? Given the underscore prefix earlier, the "public" visibility might be a mistake and it should be "protected" instead.

swiffer committed:
Update ChecksumTestFixture.php

swiffer committed:
compatibility for Postgres

swiffer commented:
@ADmad I've added compatibility for Postgres as well (https://stackoverflow.com/a/13948327).

Can you take a look please?

ADmad commented:
I don't have enough knowledge regarding Postgres to be able to review that. Hopefully someone else can.

swiffer commented:
I've tested the Statement locally, worked fine here

swiffer committed:
Update ChecksumTestFixture.php

swiffer committed:
Update ChecksumTestFixture.php

swiffer committed:
account for empty table

swiffer committed:
Update ChecksumTestFixture.php

revert add support for postgres. this does not work when a table changes that has foreign key constraints.

swiffer commented:
@ADmad I had to revert the changes for postgres because it does not work when foreign key constraints exists. this is CakePHP 4.0 compatibility only now.

jippi commented on src/TestSuite/Fixture/ChecksumTestFixture.php:
```diff
@@ -113,11 +99,13 @@ protected function _hash(ConnectionInterface $db): string
 
         if ($driver instanceof Mysql) {
             $sth = $db->execute("CHECKSUM TABLE " . $this->table . ';');
+
```
these newlines seem to be unrelated to compat with 4.x

swiffer replied:
I got this error in phpcs:

Missing blank line before return statement

jippi commented on src/TestSuite/Fixture/ChecksumTestFixture.php:
```diff
@@ -26,15 +26,9 @@ class ChecksumTestFixture extends TestFixture
     protected static $_tableHashes = [];
 
     /**
```
these comments are _not_ the same as the parent one - so these should be retained

swiffer replied:
added them back ;)

swiffer committed:
add comments back

swiffer commented:
@jippi can you take a look again so we can get this merged ?

dakota commented:
What is still needed for this?

ADmad reviewed (approved):
LGTM
